src/main/java/com/Model/ConfirmationToken.java

```diff
@@ -0,0 +1,42 @@
+package com.Model;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.persistence.*;
+import java.util.Date;
+import java.util.UUID;
+
+@Entity
+public class ConfirmationToken {
+
+    @Id
+    @GeneratedValue (strategy = GenerationType.AUTO)
+    private long id;
+
+    @Temporal(TemporalType.TIMESTAMP)
+    private Date createdDate;
+
+    @OneToOne
+    private User user;
+
+    private String confirmationToken;
```

Comment:
Can/should the token be used as a PK instead of the auto-generated id?

Reply:
Yes we should, good idea.